tests/acl/test_acl_dscp_change.py

```diff
@@ -0,0 +1,654 @@
+    '''
+        Setup for the entire script.
+        The basic steps in VxLAN configs are:
+            1. Configure VxLAN tunnel.
+            2. Configure Vnet and its VNI.
+
+            The testcases are focused on the "configure routes" step. They add,
+            delete, modify, the routes.
+    '''
+    data = {}
+
+    ptf_src_port, ptf_dst_ports, dut_port = prepare_test_port
+    data['ptfadapter'] = ptfadapter
+    data['ptf_src_port'] = ptf_src_port
+    data['ptf_dst_ports'] = ptf_dst_ports
+    data['dut_port'] = dut_port
+
+    duthost = duthosts[rand_one_dut_hostname]
+    minigraph_data = duthost.get_extended_minigraph_facts(tbinfo)
+    data['minigraph_facts'] = minigraph_data
+    data['tbinfo'] = tbinfo
+    data['duthost'] = duthost
+    if data['minigraph_facts']['minigraph_lo_interfaces'][0]['prefixlen'] == 32:
+        data['loopback_v4'] = data['minigraph_facts']['minigraph_lo_interfaces'][0]['addr']
+        data['loopback_v6'] = data['minigraph_facts']['minigraph_lo_interfaces'][1]['addr']
+    else:
+        data['loopback_v4'] = data['minigraph_facts']['minigraph_lo_interfaces'][1]['addr']
+        data['loopback_v6'] = data['minigraph_facts']['minigraph_lo_interfaces'][0]['addr']
+
+    asic_type = duthost.facts["asic_type"]
+    # enable for CISCO when support is confirmed.
+    if asic_type not in ["mellanox"]:
+        py_assert(False, "Pls update this script for your platform.")
+
+    platform = duthost.facts['platform']
+    if platform == 'x86_64-mlnx_msn2700-r0' and encap_type in ['v4_in_v6', 'v6_in_v6']:
+        pytest.skip("Skipping test. v6 underlay is not supported on Mlnx 2700")
+
+    ecmp_utils.Constants['KEEP_TEMP_FILES'] = False
+    ecmp_utils.Constants['DEBUG'] = False
+
+    Logger.info("Constants to be used in the script:%s", ecmp_utils.Constants)
+
+    data['vxlan_port'] = 4789
+    data['dut_mac'] = data['duthost'].facts['router_mac']
+
+    time.sleep(1)
+    # setting up vxlan tunnel.
+    ecmp_utils.configure_vxlan_switch(
+        data['duthost'],
+        vxlan_port=data['vxlan_port'],
+        dutmac=data['dut_mac'])
+
+    outer_layer_version = ecmp_utils.get_outer_layer_version(encap_type)
+    encap_type_data = {}
+
+    # To store the names of the tunnels, for every outer layer version.
+    tunnel_names = {}
+    # To track the vnets for every outer_layer_version.
+    vnet_af_map = {}
+
+    outer_layer_version = ecmp_utils.get_outer_layer_version(encap_type)
+    try:
+        tunnel_names[outer_layer_version]
+    except KeyError:
+        tunnel_names[outer_layer_version] = ecmp_utils.create_vxlan_tunnel(
+            data['duthost'],
+            minigraph_data,
+            af=outer_layer_version)
+
+    payload_version = ecmp_utils.get_payload_version(encap_type)
+    encap_type = "{}_in_{}".format(payload_version, outer_layer_version)
+
+    try:
+        encap_type_data['vnet_vni_map'] = vnet_af_map[outer_layer_version]
+    except KeyError:
```

Comment:
Suggest checking if `outer_layer_version` is in `vnet_af_map` and create if not.